### PR TITLE
github: call github API twice for organisation repos

### PR DESCRIPTION
**Pull request by asdine**
Currently, the Github API doesn't return internal repos
when calling it with the "all" type.
We need to call it twice, once with the "all" type and
once with the "internal" type.
However, since we don't have any guarantee that this behavior
will always remain the same and that Github will never fix this issue,
we need to deduplicate the results before sending them to the results channel.

### CODE REVIEW

File: internal/repos/github.go [hunk 1]
```diff
@@ -338,35 +338,80 @@ func (s *GithubSource) paginate(ctx context.Context, results chan *githubResult,
...
+				}
+
+				remaining, reset, retry, _ := s.v3Client.RateLimitMonitor().Get()
+				log15.Debug(
+					"github sync: ListOrgRepositories",
+					"repos", len(repos),
+					"rateLimitCost", cost,
+					"rateLimitRemaining", remaining,
+					"rateLimitReset", reset,
+					"retryAfter", retry,
```
Comment (ryanslade) [inline internal/repos/github.go:370]: Maybe we should add "type" here too?

File: internal/repos/github.go [hunk 2]
```diff
@@ -338,35 +338,80 @@ func (s *GithubSource) paginate(ctx context.Context, results chan *githubResult,
...
+		})
+
+		return oerr
+	}
+
+	go func() {
+		defer close(dedupC)
+
+		err := getReposByType("all")
+		// Handle 404 from org repos endpoint by trying user repos endpoint
```
Comment (ryanslade) [inline internal/repos/github.go:384]: Why do we need to do this?

**Review by ryanslade — APPROVED** [review]

File: internal/repos/github.go (hunk 1 above)
Comment (asdine) [reply internal/repos/github.go:370]: Yes definitely

File: internal/repos/github.go (hunk 2 above)
Comment (asdine) [reply internal/repos/github.go:384]: I don't know why we do this, I just adapted the previous behavior to this change

**Pull request merged**